modules/c/nrt/source/DateTime.c

```diff
@@ -91,6 +91,54 @@ NRTAPI(nrt_DateTime *) nrt_DateTime_fromMillis(double millis, nrt_Error * error)
     return dt;
 }
 
+NRTPRIV(NRT_BOOL) nrt_DateTime_setMonthInfoFromDayOfYear(int year,
+                                                         int dayOfYear,
+                                                         int *month,
+                                                         int *dayOfMonth)
+{
+    if (year < 1970 || year > 2037 || dayOfYear < 1 ||
+            dayOfYear > NRT_DAYS_PER_YEAR[nrtYearIndex(year)])
+    {
+        return NRT_FAILURE;
+    }
+
+    int yearIndex = nrtYearIndex(year);
+
+    /* Day falls within the first month - need a condition for this
+     * as 0 cumulative days is not included in NRT_CUMULATIVE_DAYS_PER_MONTH */
+    if (dayOfYear <= 31)
+    {
+        *month = 1;
+        *dayOfMonth = dayOfYear;
+    }
+    else
+    {
+        /* Find the entry in cumulative days per month where the day of the
+         * year fits. */
+        int monthIndex;
+        for (monthIndex = 0; monthIndex < 11; ++monthIndex)
+        {
+            int lastMonthDays =
+                    NRT_CUMULATIVE_DAYS_PER_MONTH[yearIndex][monthIndex];
```

Comment:
I think this would be a little bit cleaner, if instead of looking up lastMonthDays each loop, we instead start it at 0 and assign it to nextMonthDays at the end of the loop. If we did this, I don't think we would need the special case for January and would not need to add to our final month value.

Reply:
Updated